server/src/main/java/org/opensearch/rest/extensions/RestSendToExtensionAction.java

```diff
@@ -86,15 +91,26 @@ public RestSendToExtensionAction(
 
         List<Route> restActionsAsRoutes = new ArrayList<>();
         for (String restAction : restActionsRequest.getRestActions()) {
-            int delim = restAction.indexOf(' ');
+            Optional<String> name = Optional.empty();
+            String[] parts = restAction.split(" ");
+            if (parts.length < 2) {
+                throw new IllegalArgumentException("REST action must contain at least a REST method and route");
+            }
             try {
-                method = RestRequest.Method.valueOf(restAction.substring(0, delim));
-                path = pathPrefix + restAction.substring(delim).trim();
+                method = RestRequest.Method.valueOf(parts[0].trim());
+                path = pathPrefix + parts[1].trim();
+                if (parts.length > 2) {
+                    name = Optional.of(parts[2].trim());
```

Comment:
We don't validate action name (https://github.com/opensearch-project/OpenSearch/pull/6870/files#diff-65183b5db1b9da68efe1ea02a058bac91a2012c1fb9aa2905e1cdb81206e84cfR22), so it could contain whatever including spaces. We should either validate (no space allowed) or limit split to 3 components only (so whatever left after path is the name).

Reply:
Added validation around action naming. I think I chose some reasonable defaults, but let me know what you think.

- Cannot be null
- Cannot be empty
- Must match pattern /^[a-zA-Z0-9:/*_]*$/
- Must be <=250 characters